pegasus/sites.v3/code.org/views/learn_code_studio.haml

```diff
@@ -12,6 +12,8 @@
   else
     launch_path = "/starwars"
   end
+  company = URI.escape(request.GET['company']) if request.GET['company']
```

Comment:
We only want to set the company cookie if the company is not null.

So you should wrap the whole thing in an if block as follows:
if request.get['company']
  company = URI.escape(request.GET['company']) 
  response.set_cookie('company', {value: company, domain: ".code.org", path: '/', expires: Time.now + (365*24*3600)})
end

After you do this, please manually verify that the company cookie is not set after you hit this URL.